### PR TITLE
#192 include top

### PR DESCRIPTION
**Pull request by nwschurink**
I have added the functionality as requested in this issue: https://github.com/lukemelas/EfficientNet-PyTorch/issues/192 and this issue https://github.com/lukemelas/EfficientNet-PyTorch/issues/207
It allows passing an option during model initialisation that will omit the final layers of the model similar as to how this is implemented in the Keras version of EfficientNet.

The script below is a minimum working example of the implementation

```
from efficientnet_pytorch import EfficientNet
from efficientnet_pytorch.utils import MemoryEfficientSwish
from torchsummary import summary
from torch import nn

# A model with the final layers
model = EfficientNet.from_name("efficientnet-b0", num_classes=2, include_top=True, in_channels=1)
model.to('cuda')
summary(model,input_size=(1,100,100))

# A model without the final layers
model = EfficientNet.from_name("efficientnet-b0", num_classes=2, include_top=False, in_channels=1)
model.to('cuda')
summary(model,input_size=(1,100,100))

# A custom model build on top of the feature extraction part of EfficientNet
model = EfficientNet.from_name("efficientnet-b0", num_classes=2, include_top=False, in_channels=1)
custom_model = nn.Sequential(model,nn.Dropout(0.2),nn.Flatten(),nn.Linear(1280,100),nn.Linear(100,2),MemoryEfficientSwish())
custom_model.to('cuda')
summary(custom_model,input_size=(1,100,100))

# A custom model with pre-trained feature extraction layers
model = EfficientNet.from_name("efficientnet-b0", num_classes=2, include_top=False, in_channels=1)
custom_model = nn.Sequential(model,nn.Dropout(0.2),nn.Linear(1280,100),nn.Linear(100,2),MemoryEfficientSwish())
custom_model.to('cuda')
summary(model,input_size=(1,100,100))
```

### CODE REVIEW

**Comment by nwschurink** [comment]
@lukemelas is this a functionality that you'd like to add? It would allow people to use Efficientnet as a feature extractor, which will increase it's use case

**Comment by lukemelas** [comment]
Yes, thank you for the PR! I will merge it in the next update.

**Comment by Amiiirali** [comment]
Thanks for the PR, but I have a note:
Why we define those last layers when we are not using them? They appear in the parameters, so I think a better way would be not to define them at all when include_top=False in the __init__ part.

**Comment by deepwonder** [comment]
> Thanks for the PR, but I have a note:
> Why we define those last layers when we are not using them? They appear in the parameters, so I think a better way would be not to define them at all when include_top=False in the **init** part.

And, the output of print(model) after using "include_top=False" is still confusing. As the last layers are also printed:
[  (_bn1): BatchNorm2d(1280, eps=0.001, momentum=0.010000000000000009, affine=True, track_running_stats=True)
  (_avg_pooling): AdaptiveAvgPool2d(output_size=1)
  (_dropout): Dropout(p=0.2, inplace=False)
  (_fc): Linear(in_features=1280, out_features=3, bias=True)
  (_swish): MemoryEfficientSwish()](url)
